Shopify/polaris-react
# [v4] Export TestProvider with defaults and fix typescript react issue

### WHY are these changes introduced?

Apps consuming polaris don't have a way to use our internal contexts

There's a types issue when reacting context

### WHAT is this pull request doing?

* Adding defaults to test provider
* Exporting it
* Fix type issues
* export theme provider
* export internal contexts :( see below
* adding a named export to rollup so we can export from `test-utilities/react-testing`
* move within content context to `src/utilities`
* optional strict mode for test provider and default props
